File: Packets/packets/SwitchTeam.cs
using System;
using System.Text;


public class SwitchTeam : Packet<PacketType>
{
    public string PlayerId { get; private set; }

    public SwitchTeam(byte[] bytes) : base(bytes) { }

    public SwitchTeam(string playerId, PacketAction action) : base(PacketType.SwitchTeam, action)
    {
        this.PlayerId = playerId;
    }

    protected override void ReadPayload(byte[] bytes, byte offset)
    {
        PlayerId = Encoding.UTF8.GetString(bytes, offset, 8);
    }

    protected override void WritePayload(byte[] bytes, byte offset)
    {
        Array.Copy(Encoding.UTF8.GetBytes(PlayerId), 0, bytes, offset, 8);
    }

    public override string ToString()
    {
        return $"SwitchTeam{{ playerId='{PlayerId}' }}";
    } 
}

File: src/main/java/net/nerfatg/proxy/packet/packets/SwitchTeam.java
package net.nerfatg.proxy.packet.packets;

import net.nerfatg.proxy.packet.Packet;
import net.nerfatg.proxy.packet.PacketAction;
import net.nerfatg.proxy.packet.PacketType;

import java.nio.BufferOverflowException;
import java.nio.ByteBuffer;
import java.nio.charset.StandardCharsets;

public class SwitchTeam extends Packet {
    private String playerId;

    public SwitchTeam(ByteBuffer buffer) throws BufferOverflowException {
        super(buffer);
    }

    public SwitchTeam(String playerId, PacketAction action) {
        super(PacketType.SwitchTeam, action);
        this.playerId = playerId;
    }

    @Override
    public void readPayload(ByteBuffer buffer, int offset) throws BufferOverflowException {
        buffer.position(offset);
        byte[] playerIdBytes = new byte[8];
        buffer.get(playerIdBytes);
        this.playerId = new String(playerIdBytes, StandardCharsets.UTF_8);
    }

    @Override
    public void writePayload(ByteBuffer buffer, int offset) throws BufferOverflowException {
        buffer.position(offset);
        
        // Write playerId (8 bytes fixed)
        byte[] originalPlayerIdBytes = playerId.getBytes(StandardCharsets.UTF_8);
        buffer.put(originalPlayerIdBytes, 0, Math.min(originalPlayerIdBytes.length, 8));
        if (originalPlayerIdBytes.length < 8) {
            buffer.put(new byte[8 - originalPlayerIdBytes.length]); // pad with zeros
        }
    }

    public String getPlayerId() { return playerId; }

    @Override
    public String toString() {
        return String.format("SwitchTeam{ playerId='%s' }", playerId);
    }
}